File: peuler/ex9/ex9.py
def isPyth(a, b, c):
    return ((a*a + b*b == c*c) or
            (a*a + c*c == b*b) or
            (b*b + c*c == a*a))

for x in range(1, 1000):
    for y in range(1, 1000):
        for z in range(1, 1000):
            if (isPyth(x,y,z) and (x + y + z == 1000)):
                print(str(x*y*z) + " is the product of the Pythagorean triple with sum 1000.")
                print("This is made of: (" + str(x) + ", " + str(y) + ", " + str(z) + ").")
                quit()

File: peuler/ex9/ex9.c
#include<stdio.h>

int isPyth(int a, int b, int c) {
  return ((a * a + b * b == c * c) ||
		  (a * a + c * c == b * b) ||
		  (b * b + c * c == a * a));
}

int main() {

  for (int x = 1; x <= 1000; x++)
	for (int y = 1; y <= 1000; y++)
	  for (int z = 1; z <= 1000; z++)
		if (isPyth(x, y, z) && (x + y + z == 1000)) {
		  printf("%d is the product of the Pythagorean triple with sum 1000.\n", x * y * z);
		  printf("This is made of (%d, %d, %d).\n", x, y, z);
		  return 0;
		}
}
